crates/storage/codecs/src/alloy/transaction/eip1559.rs

```diff
@@ -0,0 +1,67 @@
+use crate::Compact;
+use alloy_consensus::TxEip1559 as AlloyTxEip1559;
+use alloy_eips::eip2930::AccessList;
+use alloy_primitives::{Bytes, ChainId, TxKind, U256};
+use serde::{Deserialize, Serialize};
+
+/// [EIP-1559 Transaction](https://eips.ethereum.org/EIPS/eip-1559)
+///
+/// This is a helper type to use derive on it instead of manually managing `bitfield`.
+///
+/// By deriving `Compact` here, any future changes or enhancements to the `Compact` derive
+/// will automatically apply to this type.
+///
+/// Notice: Make sure this struct is 1:1 with [`alloy_consensus::transaction::TxEip1559`]
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Compact, Default, Serialize, Deserialize)]
+#[cfg_attr(test, derive(arbitrary::Arbitrary))]
+#[cfg_attr(test, crate::add_arbitrary_tests(compact))]
+pub(crate) struct TxEip1559 {
+    chain_id: ChainId,
+    nonce: u64,
+    gas_limit: u64,
```

Comment:
```suggestion
    #[cfg_attr(feature = "serde", serde(with = "alloy_serde::quantity"))]
    gas_limit: u128,
```
why can't we do this?